app/controllers/PreviousMovementRecordsController.scala

```diff
@@ -18,50 +18,38 @@ package controllers
 
 import connectors.GoodsRecordConnector
 import controllers.actions._
-import pages.PreviousMovementRecordsPage
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import queries.GetGoodsRecordsQuery
-import repositories.SessionRepository
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
-import views.html.{GoodsRecordsEmptyView, PreviousMovementRecordsView}
+import views.html.PreviousMovementRecordsView
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class PreviousMovementRecordsController @Inject() (
   override val messagesApi: MessagesApi,
-  sessionRepository: SessionRepository,
   identify: IdentifierAction,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
   val controllerComponents: MessagesControllerComponents,
   view: PreviousMovementRecordsView,
-  emptyView: GoodsRecordsEmptyView,
-  getGoodsRecordConnector: GoodsRecordConnector
+  goodsRecordConnector: GoodsRecordConnector
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    getGoodsRecordConnector.doRecordsExist(request.eori).map {
-      case Some(getGoodsRecordResponse) if getGoodsRecordResponse.goodsItemRecords.nonEmpty =>
-        Ok(view())
-
-      case Some(_) =>
-        Redirect(routes.GoodsRecordsController.onPageLoadNoRecords())
+    goodsRecordConnector.getRecordsCount(request.eori).flatMap {
+      case 0 => Future.successful(Redirect(routes.GoodsRecordsController.onPageLoadNoRecords()))
+      case _ =>
+        goodsRecordConnector.doRecordsExist(request.eori).map {
+          case false => Ok(view())
+          case true  => Redirect(routes.GoodsRecordsController.onPageLoad(1))
```

Comment:
Perhaps make the 1 a "firstGoodsPage" variable?

Reply:
has the design changed for this? My understanding is that it should always go through previous movement records page if records exist and redirect happens on submit button.

Reply:
so if they have already been stored it should redirect and not see this page at all